components/script/dom/globalscope.rs

```diff
@@ -2520,7 +2520,7 @@ impl GlobalScope {
         }
         if let Some(worker) = self.downcast::<WorkerGlobalScope>() {
             // https://html.spec.whatwg.org/multipage/#script-settings-for-workers:api-base-url
-            return worker.get_url().clone();
+            return worker.get_api_base_url().clone();
```

Comment:
It's strange that, despite the spec saying we should use the URL, it also says we should use the API base url: https://html.spec.whatwg.org/multipage/webappapis.html#api-base-url and then here it only states the global scope's url. Yet in the blob case, the URL is different from the base URL and WPT tests assert for the latter. I think this is a spec bug, and https://html.spec.whatwg.org/multipage/workers.html#script-settings-for-workers%3Aapi-base-url should handle the blob case.

Reply:
Opted to keep the special new field in `WorkerGlobalScope` to make resolving relative to the parent document possible, but revert the other changes. Added WPT tests to catch these mistakes.